### ISSUE TITLE
Compartments for shelled bins

### ISSUE THREAD

**Issue by trohnjavolta** [Le0Michine/FusionGridfinityGenerator#59]
Hi, first of all thx for this, it's awesome!

I already read that with shelled bins the compartments option isn't possible but since it says "currently" I guess it's not ruled out :)
Is there any ETA when compartments will be possible for shelled bins?

**Comment by Le0Michine** [comment]
Hi, that's not very high on my priorities list. The reason being is that it's quite hard to adapt the way compartments are generated at the moment (they are subtracted from a solid body representing internal space of the bin). And in combination with scoop fillet I'm not sure "shelled" part would make sense anymore as there won't be much difference in neither capacity nor material comparing to regular bins

**Comment by trohnjavolta** [comment]
Too bad, but I can imagine it wouldn't be easy to implement.
Iwas fiddling around in fusion earlier and after deleting a few surfaces I could add a rounding:
https://ibb.co/Wk91zt1

I'm printing it right now, I'm just trying different bins out and see which ones I like best.

**Comment by Nickduino** [comment]
World it be possible to grey out an option if it's unavailable?

**Comment by Nickduino** [comment]
> Hi, that's not very high on my priorities list. The reason being is that it's quite hard to adapt the way compartments are generated at the moment (they are subtracted from a solid body representing internal space of the bin)

I just tested something and it worked. I generated a bin with compartments with your script, suppressed the bottom fillets of the compartments (I removed them from the bodies that were used for the cuts) and then shelled the finished body from the bottom of the compartments (leaving one as is to show what I started with).

And voilà:

![image](https://github.com/Le0Michine/FusionGridfinityGenerator/assets/9784237/d17b86da-cb0d-4faf-90b5-6ef6014a2160)


I personally think it's worth it to implement it (so, if shelled is ticked, don't fillet but shell the bottom of the bins) as a hollow box uses more filament (40% more on my test _without_ compartments) and take longer to print (20% longer on my test _without_ compartments) than a shelled one.

**Comment by Nickduino** [comment]
> so, if shelled is ticked, don't fillet

![image](https://github.com/Le0Michine/FusionGridfinityGenerator/assets/9784237/c367be94-337e-4a95-bee8-3e413aceda79)
So just don't pass the `input.hasBottomFillet` argument as `True` when calling binBodyCutoutGenerator.py for a bin that is shelled and has compartments ?


> but shell the bottom of the bins

Just pass the right faces to shellUtils.py?

**Comment by Le0Michine** [comment]
How does it behave when compartments don't match base grid?

**Comment by Nickduino** [comment]
> How does it behave when compartments don't match base grid?

You're right, it's less than optimal, as the walls hover over the bottom:

![image](https://github.com/Le0Michine/FusionGridfinityGenerator/assets/9784237/be980221-7963-4715-9532-f1886acea9d8)

![image](https://github.com/Le0Michine/FusionGridfinityGenerator/assets/9784237/18be5928-b0fb-4f32-b99b-9e8c02159bcd)

Still better than nothing but not great.

**Comment by Nickduino** [comment]
Here's an alternative approach that should work and not be too complex to implement:

Currently, from the input grid, you create bodies representing the compartments and you subtract them from the box. You could keep most of that math to calculate the section of the compartments, subtract them from the section of the box to **get the section of the walls and then extrude the walls to object**:

![image](https://github.com/Le0Michine/FusionGridfinityGenerator/assets/9784237/61829c9e-5092-4599-9451-602701482cf7)

**Comment by Pytonballoon810** [comment]
Is this still being worked on? I would really love this, even if it's just for compartments that match the grid…
